modules/authorizedkeys/authorizedkeys.go

```diff
@@ -1,196 +1,182 @@
+	}
+
+	return nil
 }
 
-func (r *run) openDestFiles(userkeys map[string][]string) (destfiles map[string]*os.File) {
-	var err error
-	destfiles = make(map[string]*os.File)
-	for uid, _ := range userkeys {
-		dest := interpolate(r.p.Destination, uid)
-		if _, ok := destfiles[dest]; !ok {
-			if !r.Conf.ApplyChanges {
-				log.Println("[dryrun] would have created", dest)
-				continue
-			}
-			os.MkdirAll(filepath.Dir(dest), 0750)
-			destfiles[dest], err = os.Create(dest)
+func (akm *AuthorizedKeysModule) verifyPaths() ([]string, error) {
+	if !strings.Contains(akm.config.Path, "{username}") {
+		log.Warnf("Authorized Keys 'verify' does not operate on paths without '{username}'. Skipping %s", akm.config.Path)
+		return nil, nil
+	}
+
+	globber, err := interpolate(akm.config.Path, "*")
+	if err != nil {
+		return nil, err
+	}
+	files, err := filepath.Glob(globber)
+	if err != nil {
+		return nil, err
+	}
+	fileMap := map[string]bool{}
+	for _, f := range files {
+		fileMap[f] = true
+	}
+
+	allLdapUsers, err := akm.PersonClient.GetAllActiveStaff()
+	if err != nil {
+		log.Errorf("Error getting users from Person API: %s", err)
+		return nil, err
+	}
+
+	for _, person := range allLdapUsers {
+		if akm.hasAllowedGroup(person) {
+			personPath, err := interpolate(akm.config.Path, akm.LDAPUsernameToLocalUsername(person.GetLDAPUsername(), akm.config.UsernameMap))
 			if err != nil {
-				log.Printf("[warning] can't create %q: %v", dest, err)
+				log.Errorf("Error interpolating person's (%s) file path.", person.GetLDAPUsername())
+				return nil, err
 			}
-			log.Printf("[info] creating %q", dest)
+			fileMap[personPath] = false
 		}
 	}
-	return
-}
 
-func (r *run) writePubKeys(userkeys map[string][]string, destfiles map[string]*os.File) {
-	var uids []string
-	for uid, _ := range userkeys {
-		uids = append(uids, uid)
-	}
-	sort.Strings(uids)
-	for _, uid := range uids {
-		dest := interpolate(r.p.Destination, uid)
-		if !r.Conf.ApplyChanges {
-			log.Printf("[dryrun] would have written pubkey of %q into %q", uid, dest)
-			continue
-		}
-		marker := ""
-		if r.p.DeletionMarker != "" {
-			marker = fmt.Sprintf("; deletionmarker=%s", r.p.DeletionMarker)
+	badPaths := []string{}
+	for filePath, notInLdap := range fileMap {
+		if !notInLdap {
+			badPaths = append(badPaths, filePath)
 		}
-		fmt.Fprintf(destfiles[dest], "# %s LDAP pubkeys%s\n", uid, marker)
-		if len(userkeys[uid]) == 0 {
-			fmt.Fprintf(destfiles[dest], "# no key found in ldap, ask user to add one\n")
-			continue
-		}
-		var keys []string
-		for _, key := range userkeys[uid] {
-			keys = append(keys, key)
-		}
-		sort.Strings(keys)
-		for _, key := range keys {
-			fmt.Fprintf(destfiles[dest], "%s\n", key)
-		}
-		log.Printf("[info] %d keys written into %q", len(userkeys[uid]), dest)
 	}
-	return
+
+	return badPaths, nil
 }
 
-func interpolate(orig, uid string) string {
+func interpolate(orig, uid string) (string, error) {
```

Comment:
maybe preprocess known vars then pass them to a https://golang.org/pkg/text/template/ str?